e2e/runner/poste2e_checker.go

```diff
@@ -0,0 +1,61 @@
+package runner
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/protocol-contracts/v2/pkg/zrc20.sol"
+
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// EnsureNoTrackers ensures that there are no trackers left on zetacore
+func (r *E2ERunner) EnsureNoTrackers() {
+	// get all trackers
+	res, err := r.CctxClient.OutTxTrackerAll(
+		r.Ctx,
+		&crosschaintypes.QueryAllOutboundTrackerRequest{},
+	)
+	require.NoError(r, err)
+	require.Empty(r, res.OutboundTracker, "there should be no trackers at the end of the test")
+}
+
+// EnsureZeroBalanceOnRestrictedAddress ensures that the balance of the restricted address is zero
+func (r *E2ERunner) EnsureZeroBalanceOnRestrictedAddress() {
```

Comment:
Let's make it generic.
```suggestion
func (r *E2ERunner) EnsureZeroBalanceAddress(address string) {
```

Reply:
Also

- Let's rename this file to `runner/require.go` or similar as it can be invoked at any point of E2E test.
- And let's add a comment that it applies only to zevm and not external chains

Reply:
yeah, sure thing.

Reply:
renamed in commit https://github.com/zeta-chain/node/pull/2795/commits/6e3f2dd0899a8d598c4a45d60a12d944aecf9217